### PR TITLE
feat: support for deployment on android (production)

### LINKED ISSUE #5: Add support for deploying on Android (Production)
- Add deploy step to existing CI/CD pipeline for 'main' to build and deploy on Android Play Store. The build/deploy should be taken care by fastlane.
- Add support for deploying using production config (option to use - '.env.production' file, 'config' module).

### PR DESCRIPTION
**Pull request by abhishek310001**
## Description
_This PR add support of uploading the Application to Production track of Google Play Console._

Changlelogs Support has not been added yet.

### Google Play Console 
Screenshot 2024-08-02 at 16 42 34

### GH Action 
Screenshot 2024-08-02 at 16 43 45 |

### Automated test cases added
NA

### Manual test cases run
_For each manual test case, list the steps to test or reproduce the PR._
NA

### CODE REVIEW

**Review by bbbneo333 — COMMENTED** [review]
@abhishek310001 - LGTM! Requesting following on this:
- Can you add a screenshot from play console which has the details on uploaded app (name, version number, changelog)
- What does the procedure does it follows for generating the version number and changelog?
- Can we also add any documentation / links around what steps one needs to follow in order to configure the deployment end to end?

**Comment by bbbneo333** [comment]
@abhishek310001 - Also can you tell: If I were to push a release, and there an existing release which hasn't been submitted yet - What would happen to the old release? Would it get invalidated? Or would it be still around?

**Comment by abhishek310001** [comment]
> @abhishek310001 - Also can you tell: If I were to push a release, and there an existing release which hasn't been submitted yet - What would happen to the old release? Would it get invalidated? Or would it be still around?

if there's already a release that haven't been submitted yet, then pushing a new release will superseded by the new release (attached a SS for reference). The previous release will still be accessible.
Screenshot 2024-08-02 at 16 34 39

File: .github/workflows/production_on_push.yml [hunk 1]
```diff
@@ -56,7 +56,25 @@ jobs:
...
           docker_password: ${{ secrets.DOCKER_PASSWORD }}
-  
+
+  deploy:
+    needs: build
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout (app)
+        uses: actions/checkout@v3
+        # make sure to keep checked in code at different directory to avoid colliding with docker caching
```
Comment (bbbneo333) [inline .github/workflows/production_on_push.yml]: nit: Can remove the comment now

File: README.md [hunk 2]
```diff
@@ -113,6 +113,54 @@ to `console,datadog`.
...
+    - Ensure **JSON** is selected as the Key type, and click **CREATE**.
+    - Save the file on your computer when prompted and remember its location.
+  3. Open the [Google Play Console](https://play.google.com/console/?hl=en) and select **Users and Permissions**.
+    - Click **Invite new users**.
+    - Paste the email address you saved earlier into the email address field.
+    - Click on **Account Permissions**.
+    - Choose the permissions you'd like this account to have. We recommend **Admin (all permissions)**, but you may want to manually select all checkboxes and leave out some of the **Releases** permissions, such as **Release to production, exclude devices, and use Play App Signing**.
+    - Click on **Invite User**.
+
+## Setting Up Repo Secrets
```
Comment (bbbneo333) [inline README.md]: nit:  Setting up Github Workflow

**Review by bbbneo333 — APPROVED** [review]
LGTM! Please merge after addressing the comments

**Pull request merged**